FreeAPS/Sources/Shortcuts/Override/overridePresetsIntentRequest.swift

```diff
@@ -0,0 +1,91 @@
+import CoreData
+import Foundation
+
+@available(iOS 16.0, *) final class OverridePresetsIntentRequest: BaseIntentsRequest {
+    enum overridePresetsError: Error {
+        case noTempOverrideFound
+        case noDurationDefined
+    }
+
+    var overrideList: [OverridePresets] {
+        var profileArray = [OverridePresets]()
+        let requestProfiles = OverridePresets.fetchRequest() as NSFetchRequest<OverridePresets>
+        try? profileArray = coredataContext.fetch(requestProfiles)
+        return profileArray
+    }
+
+    func fetchAll() -> [overridePreset] {
+        overrideList.compactMap { override in
+            guard let id = override.id, let name = override.name else { return nil }
+            return overridePreset(id: id, name: name)
+        }
+        // convert(tt: tempTargetsStorage.presets())
+    }
+
+    func fetchIDs(_ uuid: [overridePreset.ID]) -> [overridePreset] {
+        let UUIDTempTarget = overrideList.filter {
+            guard let id = $0.id else { return false }
+            return uuid.contains(id) }
+
+        return UUIDTempTarget.compactMap { override in
+            guard let id = override.id, let name = override.name else { return nil }
+            return overridePreset(id: id, name: name)
+        }
+    }
+
+    func fetchOne(_ uuid: overridePreset.ID) -> overridePreset? {
+        let UUIDTempTarget = overrideList.filter { uuid == $0.id! }
```

Comment:
Should the id here be gracefully unwrapped to avoid crashes in case it is `nil`?

Reply:
@dnzxy Fixed